File: app/schemas/booking.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime
from app.models.booking import BookingStatus


class BookingBase(BaseModel):
    flight_id: int
    passenger_profiles: List[dict]  # List of {passenger_profile_id, seat_number}


class BookingCreate(BookingBase):
    pass


class BookingResponse(BaseModel):
    id: int
    pnr: str
    user_id: int
    flight_id: int
    status: BookingStatus
    seat_hold_expires_at: Optional[datetime]
    created_at: datetime
    flight: dict
    tickets: List[dict]

    class Config:
        from_attributes = True


class BookingDetailResponse(BookingResponse):
    payment: Optional[dict] = None

File: frontend/lib/models/booking.dart
import 'flight.dart';

class Ticket {
  final int id;
  final String ticketNumber;
  final String seatNumber;

  Ticket({
    required this.id,
    required this.ticketNumber,
    required this.seatNumber,
  });

  factory Ticket.fromJson(Map<String, dynamic> json) {
    return Ticket(
      id: json['id'],
      ticketNumber: json['ticket_number'],
      seatNumber: json['seat_number'],
    );
  }
}

class Booking {
  final int id;
  final String pnr;
  final String status;
  final Flight flight;
  final List<Ticket> tickets;

  Booking({
    required this.id,
    required this.pnr,
    required this.status,
    required this.flight,
    required this.tickets,
  });

  factory Booking.fromJson(Map<String, dynamic> json) {
    return Booking(
      id: json['id'],
      pnr: json['pnr'],
      status: json['status'],
      flight: Flight.fromJson(json['flight']),
      tickets: (json['tickets'] as List?)
          ?.map((t) => Ticket.fromJson(t))
          .toList() ?? [],
    );
  }
}
